src/web/mjs/connectors/CarToonMad.mjs

```diff
@@ -0,0 +1,64 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class CarToonMad extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'cartoonmad';
+        super.label = 'CarToonMad';
+        this.tags = ['manga', 'chinese'];
+        this.url = 'https://www.cartoonmad.com';
+        this.requestOptions.headers.set('x-referer', this.url);
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        const request = new Request(new URL('/comic99.html', this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'td[align] > a:last-child.pages');
+        const pageCount = parseInt(data[0].href.match(/(\d+)\.html/)[1]);
+        for(let page = 1; page <= pageCount; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const request = new Request(new URL('/comic99.'+('00' + page).slice(-2)+'.html', this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'a.a1',0,'big5');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.textContent.trim()
+            };
+        });
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'td:nth-child(2)  tr:nth-child(3) > td:nth-child(2) > a:last-child',0,'big5');
+        const id = uri.pathname + uri.search;
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getChapters(manga) {
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, '#info td > a',0,'big5');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.textContent.trim()
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'body');
```

Comment:
Does this not require `big5` encoding?

Reply:
no, works fine
i also don't use it for getting the pageCount in _getMangas

Reply:
big5 is only used for the traditional chinese text everything else is regular